misc/config_tools/schema/checks/vm_types.xsd

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0"?>
+<xs:schema xml:id="root"
+	   xmlns:xs="http://www.w3.org/2001/XMLSchema"
+	   xmlns:acrn="https://projectacrn.org">
+
+  <xs:assert test="count(/acrn-config/vm[vm_type = 'SERVICE_VM']) &lt; 2">
+    <xs:annotation acrn:severity="error">
+      <xs:documentation>There can be at most one service VM, but more than one was configured. Verify there is only one VM with  vm_type set to SERVICE_VM. </xs:documentation>
+    </xs:annotation>
+  </xs:assert>
+
+  <xs:assert test="every $pre_vm in /acrn-config/vm[starts-with(vm_type, 'PRE')] satisfies
+                   every $svm in /acrn-config/vm[vm_type = 'SERVICE_VM'] satisfies
+                   $pre_vm/@id &lt; $svm/@id">
+    <xs:annotation acrn:severity="error">
+      <xs:documentation>This pre-launched VM (ID: {$pre_vm/@id}) must have a VM ID value less than the service VM (ID: {$svm/@id}). Adjust the VM ID values to fix this.</xs:documentation>
```

Comment:
How does this situation happen?  How does the developer change a VM ID in the configurator?  I don't think we have a way in the UI do do this? For example, you can't drag and drop a VM tab to move them around, right?

Reply:
See the response above.